addons/sale_expense_margin/models/sale_order_line.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    expense_id = fields.Many2one('hr.expense', string='Expense')
+
+    @api.depends('is_expense')
+    def _compute_purchase_price(self):
+        date_today = fields.Date.context_today(self.env.user)
+        expense_lines = self.filtered(lambda sol: sol.expense_id)
+        for line in expense_lines:
+            if line.expense_id.product_has_cost:
+                product_cost = line.expense_id.untaxed_amount / line.expense_id.quantity
+            else:
+                product_cost = line.expense_id.untaxed_amount
+
+            from_currency = line.expense_id.currency_id or self.env.company.currency_id
```

Comment:
```suggestion
            from_currency = line.expense_id.currency_id
```
The currency_id is required on the expenses, is the fallback really necessary ?